x-pack/plugins/security_solution/server/endpoint/lib/artifacts/lists.ts

```diff
@@ -97,10 +97,18 @@ export function translateToEndpointExceptions(
   exc: FoundExceptionListItemSchema,
   schemaVersion: string
 ): TranslatedExceptionListItem[] {
+  const entrySet = new Set();
```

Comment:
I think this would be cleaner using `reduce` with an accumulator that contains both the set and the filtered list... and then just return the list at the end. Or you could use `filter`, but using `reduce` could avoid the global state.

Reply:
I was going to use filter but avoided it because I thought it would just make things worse since I have to add to the set as well after checking if it exists.

Reply:
I think filter or reduce would be preferred over forEach here. My guess is that reduce will be the best fit, but again, I'm still learning.

Reply:
I agree with @madirey here. Using reduce as she describes would be cleaner. However, I think it would be better if we pulled out the super duper deduping logic out of this function and into a separate function. That way we could test that logic independently. `translateToEndpointExceptions` and `translateItem` should do only that, translation. Deduping might be better handled elsewhere.

Reply:
Not sure if I agree, pulling this out would still force us to keep some sort of state within the translation functions, and just move the `Set` methods elsewhere. Not much use in testing that separately I think.

I'll look into filter or reduce

Reply:
We talked and now we like this. 💯